outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
+    1. Filenames - named the same as the export
+        1. Every file exports one thing
+    2. Avoid depending on load order via namespacing and Meteor.startup
+        1. If you depend on load order, your best bet is a package
+    3. Naming guidelines for everything
+        1. https://github.com/meteor/guide/issues/9#issuecomment-148800085
+    4. Templates should have a CSS class that matches the template name, and other Blaze suggestions
+    5. Meteor ES2015 style guide (link to it, don't reproduce all of it)
+    6. Perhaps this should include copypasta boilerplate - a standard template/component, a standard publication, standard route, etc? Basically the stuff a scaffolding framework would generate for you?
+5. Small app
+    1. Directory structure around features, not client/server
```

Comment:
I'll just add one thing that might not have come across from my terse comments above:

In the past, we've made efforts to put an entire "slice" of the app in a package (e.g. in galaxy we had an `activity-feed` package that included the collection, publications, as well as the component to render the feed).

But inevitably we've always ended up splitting it into the "business logic" part (i.e. the collection, most of the more general methods + pubs, etc -- a package called `activities`) and the "view" part (the final `activity-feed` package, which depends on `activities`).

I would suggest we don't suggest ever mixing collections and templates in a package for this reason. I'm not sure if this is controversial or not.